doc/index.md

```diff
@@ -0,0 +1,154 @@
+# Documentation for mlpack
+
+## A fast, flexible machine learning library
+
+mlpack is an intuitive, fast, and flexible header-only C++ machine learning
+library with bindings to other languages.  It aims to provide fast, lightweight
+implementations of both common and cutting-edge machine learning algorithms.
+
+mlpack's lightweight C++ implementation makes it ideal for deployment, and it
+can also be used for interactive prototyping via C++ notebooks (these can be
+seen in action on mlpack's [homepage](https://www.mlpack.org/)).
+
+In addition to its powerful C++ interface, mlpack also provides command-line
+programs, and bindings to the Python, R, Julia, and Go languages.
+
+_If you use mlpack, please [cite the software](citation.md)._
+
+## mlpack basics
+
+Installing mlpack can be done using the
+[instructions in the README](README.md#3-installing-and-using-mlpack-in-c);
+or the [Windows build guide](user/build_windows.md).
+The following basic guides are *highly recommended* before using mlpack.
+
+ * ***First steps***:
+   - [mlpack C++ quickstart](quickstart/cpp.md): create a couple simple C++
+     programs that use mlpack
+   - [Sample Windows mlpack C++ application](user/sample_ml_app.md): create a
+     working mlpack Windows program using Visual Studio
+
+ * ***Basics of matrices and data in mlpack***:
+   - [Matrices and data in mlpack](user/matrices.md)
+   - [Loading and saving mlpack objects](user/load_save.md)
+
+ * ***Reference for mlpack core classes***:
+   - [Core mlpack documentation](user/core.md)
+
+ * ***Bindings to other languages*** (Python/R/CLI/Julia/Go):
```

Comment:
```suggestion
 * ***Using mlpack natively with our extensions in Python, R, CLI, Julia, and Go *** :
```

Bindings is probably a bit too abstract. This emphasizes more an enhancement.

Reply:
Thanks, this is a nice improvement. :+1: